models/inventory/StockMovementItem.ts

```diff
@@ -236,6 +237,7 @@ export class StockMovementItem extends Doc {
 
   override hidden: HiddenMap = {
     batch: () => !this.fyo.singles.InventorySettings?.enableBatches,
+    serialNo: () => !this.fyo.singles.InventorySettings?.enableSerialNo,
```

Comment:
Should be visible only if the selected item has Serial Nos enabled (same with batches).